src/runtime_src/core/common/api/aie/xrt_graph.cpp

```diff
@@ -112,6 +112,50 @@ class graph_impl
 
 }
 
+namespace xrt { namespace aie {
+
+class event_impl
+{
+private:
+  std::shared_ptr<xrt_core::device> device;
+  int handle;
+
+public:
+  event_impl(std::shared_ptr<xrt_core::device> dev, int phdl)
+    : device(std::move(dev))
+    , handle(phdl)
+  {}
+
+  ~event_impl()
+  {}
+
+  uint64_t
+  read_profiling()
+  {
+    return device->read_profiling(handle);
+  }
+
+  void
+  stop_profiling()
+  {
+    device->stop_profiling(handle);
+  }
+
+  int getHandle() const { return handle; }
+};
+
+}
+
+}
+
+int get_event_handle(const std::shared_ptr<xrt::aie::event_impl>& event_ptr) {
```

Comment:
Why is this extra function required? you can directly get handle using event_ptr->getHandle() function.